raco/backends/myria/myria.py

```diff
@@ -413,10 +413,40 @@ def compileme(self, leftid, rightid):
             "argSelect1": allleft,
             "argSelect2": allright
         }
+        if hasattr(self, 'pull_order') and self.pull_order is not None:
+            join.update({"argOrder": self.pull_order})
 
         return join
 
 
+class MyriaIDBController(algebra.IDBController, MyriaOperator):
+
+    def compileme(self, *args):
+        ret = {
+            "opType": "IDBController",
+            "argSelfIdbId": "%s" % self.idb_id,
+            "argInitialInput": "%s" % args[0],
+            "argIterationInput": "%s" % args[1],
+            "argEosControllerInput": "%s" % args[2],
+            "argState": self.get_agg_json(),
+            "sync": self.recursion_mode == "SYNC"
```

Comment:
The initial purpose of adding `IDBController` was for async iterations, since `DoWhile` can handle sync already (although it has to store after each iteration). Then we added the sync part to `IDBController` since it's just a small change that fits into the operator design and we think it's convenient to have both in one operator. In terms of worst/best-case performance... we can always make things happen :)